README.md

```diff
@@ -28,4 +28,294 @@ One of the very common use cases of WebXR is 180 and/or 360 photos or videos, bo
+
+```js
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+
+const layer = xrSession.requestLayer('quad');
+const isStereo = true;
+const source = await xrGfx.setFramebufferLayerSource(layer, 1024, 768, { stereo: true });
+```
+
+Passing `true` for stereo here indicates that you are able to provide stereo imagery for this layer source, but if the XR device is unable to display stereo imagery it may automatically force the layer to be created as mono instead to reduce memory and rendering overhead. Layers that are created as mono will never be automatically changed to stereo, regardless of hardware capabilities.
+
+## Rendering
+
+During `XRFrame` processing each layer source can be updated with new imagery. Calling `getViewSubImage()` with a view from the `XRFrame` will return an `XRSubImage` indicating what section of the associated graphics resources will be presented to the view's associated physical display. If a layer source has not yet been set for the layer `getSubImage()` with return an `XRSubImage` with all values set to zero or false.
+
+Every layer source will also have a `getCurrent_____()` method that retrieves the graphics resources that should be rendered to for the current `XRFrame`. Calling this method indicates that the imagery should be updated, otherwise any previous rendering will be displayed again, reprojected if necessary.
+
+```js
+// Render Loop for a projection layer with a WebGL framebuffer source.
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+
+const layer = xrSession.requestLayer('projection');
+const source = await xrGfx.setFramebufferLayerSource(layer);
+
+xrSession.updateRenderState({ layers: [layer] });
+xrSession.requestAnimationFrame(onXRFrame);
+
+function onXRFrame(time, xrFrame) {
+  xrSession.requestAnimationFrame(onXRFrame);
+
+  gl.bindFramebuffer(gl.FRAMEBUFFER, source.getCurrentFramebuffer());
+  for (let view in xrViewerPose.views) {
+    let viewport = layer.getViewSubImage(view).viewport;
+    gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+    
+    // Render from the viewpoint of xrView
+  }
+}
+```
+
+In some cases, such as a mono `XRQuadLayer` being shown on a stereo device, multiple `XRView`s may return the same `XRSubImage` values. To avoid rendering the same view multiple times in these scenarios the `primary` attribute of the `XRSubImage` will be be set to `false` for all but one of the overlapping views. (It should be noted that `XRProjectionLayer`s will never return `XRSubImage`s with `primary` set to false.)
```

Comment:
I think mono layers should have a different method since they are typically not dependent on the view

Reply:
What is more interesting is the multiview case, where you render both views in one pass. We need a way to handle this.

Reply:
you will be able to do this. There is nothing stopping you from looking at both subimages at the same time.

Reply:
I was a bit confused here. So, 'primary' is the way to check for mono, but it is not really working for mutliview. One thing when you render one view (mono), and another thing when you render multiview (stereo, both views, but in one pass).

I think we just need a sample for mutliview case here, something like:

```js
// Render Loop for a projection layer with a WebGL framebuffer source.
const xrGfx = new XRWebGL2GraphicsBinding(xrSession, gl);
const layer = xrSession.requestLayer('projection');
const source = await xrGfx.setTextureArrayLayerSource(layer);
const mv_ext = gl.getExtension('OVR_multiview2');
xrSession.updateRenderState({ layers: [layer] });
xrSession.requestAnimationFrame(onXRFrame);

let fbo = gl.createFramebuffer();

function onXRFrame(time, xrFrame) {
  xrSession.requestAnimationFrame(onXRFrame);
  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
  if (mv_ext != null) {
    // multiview rendering
    let colorTexture = source.getCurrentColorTexture()
    mv_ext.framebufferTextureMultiviewOVR(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, colorTexture, 0, 0, 2);
    let leftView = xrViewerPose.views[0];
    let rightView = xrViewerPose.views[1];
    let leftProj = leftView.projectionMatrix;
    let rightProj = rightView.projectionMatrix;
    let viewport = layer.getViewSubImage(leftView).viewport; // viewports should be the same for left & right eyes
    gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
    ....
  } else {
    // non-multiview rendering, using texture array
    for (let view in xrViewerPose.views) {
      let viewport = layer.getViewSubImage(view).viewport;
      gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
    
      // Render from the viewpoint of xrView
    }
  }
}
```